rpc/block.go

```diff
@@ -361,3 +361,43 @@ func nilToZero(f *felt.Felt) *felt.Felt {
 	}
 	return f
 }
+
+func (h *Handler) JunoGetBlockWithTxsAndReceipts(id BlockID) (*BlockWithTxs, *BlockWithReceipts, *jsonrpc.Error) {
+	block, rpcErr := h.blockByID(&id)
+	if rpcErr != nil {
+		return nil, nil, rpcErr
+	}
+
+	blockStatus, rpcErr := h.blockStatus(id, block)
+	if rpcErr != nil {
+		return nil, nil, rpcErr
+	}
+
+	finalityStatus := TxnAcceptedOnL2
+	if blockStatus == BlockAcceptedL1 {
+		finalityStatus = TxnAcceptedOnL1
+	}
+
+	txs := make([]*Transaction, len(block.Transactions))
+	txsWithReceipts := make([]TransactionWithReceipt, len(block.Transactions))
+	for index, txn := range block.Transactions {
+		txs[index] = AdaptTransaction(txn)
+		r := block.Receipts[index]
+		txsWithReceipts[index] = TransactionWithReceipt{
+			Transaction: txs[index],
+			Receipt:     AdaptReceipt(r, txn, finalityStatus, nil, 0, false),
+		}
+	}
+
+	return &BlockWithTxs{
+			Status:       blockStatus,
+			BlockHeader:  adaptBlockHeader(block.Header),
+			Transactions: txs,
+		},
+		&BlockWithReceipts{
+			Status:       blockStatus,
+			BlockHeader:  adaptBlockHeader(block.Header),
+			Transactions: txsWithReceipts,
+		},
+		nil
```

Comment:
Can you combine this into a single type? We are returning Status and BlockHeader twice